app/tools/jwtToken.js

```diff
@@ -0,0 +1,31 @@
+const jwt = require('jsonwebtoken'),
+  JWT_KEY = require('../constants').jwt_key,
+  moment = require('moment');
+
+  const defaultExpiration = () => moment().add(moment.duration(parseInt(process.env.SESSION_EXP), 'seconds'));
+
+exports.createToken = (payload, exp = defaultExpiration()) => {
+  return jwt.sign({ data: payload, exp: exp.valueOf() }, JWT_KEY)
+};
+
+exports.verifyToken = token => {
+  return new Promise((resolve, reject) => {
+    jwt.verify(token, JWT_KEY, (jwtError, decoded) => {
+      if (decoded) {
+        resolve(decoded);
+      } else {
+        reject(jwtError);
+      }
+    });
+  });
+};
+
+exports.veryfyExpiration = (payload, comparisonTimestamp = moment()) => {
```

Comment:
i wont reject a promise for this. just return the value and eval with a if, or return a boolean but resolving. Promises should fail and enter in a catch when the external dependency fails

Reply:
resolving an object with value and error should be fine?

Reply:
yes!